### PR TITLE
Euler vs quaternion

### PR DESCRIPTION
**Pull request by JuanMatSa**
Still a lot of things to implement, like tests, docstrings... at least the equations to transform quaternions to euler angles (and euler angles to quaternions) are implemented.

### CODE REVIEW

**Comment by JuanMatSa** [comment]
What do you think about the check? as far as I know, np.isclose is not implemented for numpy 1.6... should we have it into consideration?

**Comment by Juanlu001** [comment]
By no means. The 1.6.x series of NumPy is incredibly buggy and 1.6.2 was released almost four years ago. With modern packaging tools it makes no sense to support such an old version, so it's safe to say we strive for NumPy 1.8 or even NumPy 1.9.

**Comment by JuanMatSa** [comment]
Copy that!
Some companies still use 1.6 ¬¬

**Comment by Juanlu001** [comment]
I hear your concerns. Paraphrasing Nick Coghlan, "Stop Supporting NumPy 1.6 (For Free)" :wink: 

http://www.curiousefficiency.org/posts/2015/04/stop-supporting-python26.html

**Comment by Juanlu001** [comment]
Quote from the article:

> if you're currently supporting Python 2.6 for free because folks using RHEL 6 or one of its derivatives say they need it, and this is proving to be a hassle for you, then stop. If they complain, then point them at this post, as providing an easily linkable reference for that purpose is one of the main reasons I'm writing it. If they still don't like it, then you may want to further suggest that they come argue with me about it, and leave you alone.

The point here is that might be good reasons for big companies to have ancient technology stacks, but that there are no reasons for us to support those for free in our volunteer time. So yeah, if [random aeronautical multinational company] wants to use this software with Python 2.6 and NumPy 1.6 on Windows XP, they can hire us :smile:

**Comment by JuanMatSa** [comment]
que opinais de la nomenclatura?

obviamente euler_angles, quaternion(que tiene sus cuatro componentes dentro:  q_0, q_1, q_2, q_3), quaternion_vel que es la derivada del cuaternion y los elementos de quaternion_vel sean d_q0, dq_1, dq_2, dq_3

**Comment by AlexS12** [comment]
Esta semana buscamos un día para mirar esto juntos, vale?

**Comment by AlexS12** [comment]
Sería bastante interesante hacer merge de esta rama, ya que en #65 tengo contemplado que el sistema mantenga los quaterniones actualizados aunque trabaje con ángulos de euler para integrar. La verdad es que no recuerdo muy bien el estado de esto. @JuanMatSa nos puedes comentar algo? estaba para revisar ya?

**Comment by Juanlu001** [comment]
@JuanMatSa dinos en qué estado está esto y @AlexS12 si se puede postponer para 0.2 lo dejamos pendiente.

**Comment by AlexS12** [comment]
@JuanMatSa told me that he is no longer participating in the project, at least for the next months.

It is my intention to merge this, but I would like someone to review the code and I have my own comments. hands up :hand:?? @AeroPython/pyfme  

@Juanlu001, what is the best approach for a "dead" pull request? should I clone the branch, and start a new pull request and close this one?

**Comment by Juanlu001** [comment]
Yes, do exactly that! Assuming you cannot already push to his branch.

**Comment by AlexS12** [comment]
Maybe we can incorporate some ideas from this repo: https://github.com/satellogic/quaternions

File: src/pyfme/utils/change_euler_quaternion.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    References
+    ----------
+    .. [1] "Modeling and Simulation of Aerospace Vehicle Dynamics" (Aiaa\
+        Education Series) Peter H. Ziepfel
+    '''
+    check_unitnorm(quaternion)
+
+    q_0, q_1, q_2, q_3 = quaternion
+
+    check_unitnorm(quaternion)
```
Comment (AlexS12) [inline src/pyfme/utils/change_euler_quaternion.py]: Maybe this is too much if it has to be evaluated each time step. But I think we'll have time to get rid of it later

**Pull request merged**